src/storage/src/hummock/compactor/compaction_utils.rs

```diff
@@ -35,13 +35,13 @@ use crate::hummock::compactor::{
 use crate::hummock::multi_builder::TableBuilderFactory;
 use crate::hummock::sstable::DEFAULT_ENTRY_SIZE;
 use crate::hummock::{
-    CachePolicy, FilterBuilder, HummockResult, MemoryLimiter, SstableBuilder,
-    SstableBuilderOptions, SstableObjectIdManagerRef, SstableWriterFactory, SstableWriterOptions,
+    CachePolicy, FilterBuilder, GetObjectId, HummockResult, MemoryLimiter, SstableBuilder,
+    SstableBuilderOptions, SstableWriterFactory, SstableWriterOptions,
 };
 use crate::monitor::StoreLocalStatistic;
 
 pub struct RemoteBuilderFactory<W: SstableWriterFactory, F: FilterBuilder> {
-    pub sstable_object_id_manager: SstableObjectIdManagerRef,
+    pub sstable_object_id_manager: Box<dyn GetObjectId>,
```

Comment:
nits: better change the variable name to `object_id_getter` or change the trait name to `SstableObjectIdManager`.